Enable emitting Honeycomb events for ATC requests

njbennett opened the pull request:
If CONCOURSE_HONEYCOMB_API_KEY is set, we will wrap all endpoints with
a handler that attempts to emit Honeycomb events. If the API_KEY is not
valid event emission should silently fail.

By default, events will be sent a "concourse" dataset.
Set CONCOURSE_HONEYCOMB_DATASET to override this value.

For more information about Honeyomb, see the [honeycomb quick start
guide.](https://docs.honeycomb.io/learning-about-observability/quick-start/)

# Existing Issue
This is a start at implementing the feature request in issue #3958

# Why do we need this PR?
This PR will allow Concourse operators to understand incoming traffic to their Concourse installations in more detail. It's also a step towards instrumenting Concourse's internal container allocation with Honeycomb events, which the PAS RelEng team hopes will help us understand mysterious failures of, for instance, resources triggering.

# Changes proposed in this pull request
Adds a Honeycomb handler that emits some info about requests
If a Honeycomb client exists, wrap all endpoints with that handler
Get configuration from ENV variables

# Contributor Checklist
- [x] Unit tests
- [ ] Updated documentation (located at https://github.com/concourse/docs)
- [ ] Updated release notes (located at https://github.com/concourse/concourse/tree/master/release-notes)

(Haven't updated documentation and release notes but I'm happy to do that next Friday if this PR is otherwise acceptable. Or to hold off on adding documentation until the feature is more complete. It may or may not be useful in its current state -- we'll want to deploy it in our RelEng CI environment and explore the data it creates.)

# Reviewer Checklist
- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed

xtremerui commented:
@cirocosta

xtremerui commented:
Hi @njbennett , with the completed #4607, do you still want this PR to be merged? Thx!

njbennett commented:
Because it doesn't support annotations/fields, it doesn't look like #4607 addresses the problem that I was interested in.

However, in the 3 months since I opened this PR, I've changed teams and moved on to other problems, so I wouldn't be able to do the follow up work for this PR. You should delete it rather than merging it, it's gone stale.
